consensus/polybft/consensus_runtime.go

```diff
@@ -277,26 +271,46 @@ func (c *consensusRuntime) FSM() (*fsm, error) {
 				return nil, err
 			}
 
-			if len(nonExecutedCommitments) > 0 {
-				bundlesToExecute, err := c.state.getBundles(nextStateSyncExecutionIdx, maxBundlesPerSprint)
-				if err != nil {
-					return nil, err
+			var bundleProofs []*BundleProof
+
+			var commitments []*CommitmentMessageSigned
+
+			processedBundles := 0
+			foundGap := false
+
+			for _, c := range nonExecutedCommitments {
```

Comment:
do we have tests for it?

Reply:
Yes, but those tests need to be integrated in edge through a separate PR, once we get the v3-contracts integration to edge as well.